test/runs.test.js

```diff
@@ -377,3 +379,76 @@ describe('Run methods', () => {
         });
     });
 });
+
+describe('Redirect run logs', () => {
+    let baseUrl;
+
+    beforeAll(async () => {
+        // Ensure that the tests that use characters like á are correctly decoded in console.
+        process.stdout.setDefaultEncoding('utf8');
+        const server = await mockServer.start();
+        baseUrl = `http://localhost:${server.address().port}`;
+    });
+
+    afterAll(async () => {
+        await Promise.all([mockServer.close()]);
+    });
+
+    let client;
+    beforeEach(async () => {
+        client = new ApifyClient({
+            baseUrl,
+            maxRetries: 0,
+            ...DEFAULT_OPTIONS,
+        });
+    });
+    afterEach(async () => {
+        client = null;
+    });
+
+    describe('run.getStreamedLog', () => {
+        test('getStreamedLog - fromStart', async () => {
+            const logSpy = jest.spyOn(LoggerActorRedirect.prototype, '_console_log').mockImplementation(() => {});
+
+            // Set fake time in constructor to skip the first redirected log entry, fromStart=True should redirect all logs
+            jest.useFakeTimers();
+            jest.setSystemTime(new Date('2025-05-13T07:24:12.686Z'));
+            const streamedLog = await client.run('redirect-run-id').getStreamedLog();
+            jest.useRealTimers();
+
+            streamedLog.start();
+            // Wait some time to accumulate logs
+            await new Promise((resolve) => {
+                setTimeout(resolve, 1000);
+            });
+            await streamedLog.stop();
+
+            const loggerPrefix = c.cyan('redirect-actor-name runId:redirect-run-id -> ');
+            expect(logSpy.mock.calls).toEqual(MOCKED_ACTOR_LOGS_PROCESSED.map((item) => [loggerPrefix + item]));
+            logSpy.mockRestore();
+        });
+
+        test('getStreamedLog - not fromStart', async () => {
+            const logSpy = jest.spyOn(LoggerActorRedirect.prototype, '_console_log').mockImplementation(() => {});
+
+            // Set fake time in constructor to skip the first redirected log entry, fromStart is redirecting only new logs
+            jest.useFakeTimers();
+            jest.setSystemTime(new Date('2025-05-13T07:24:12.686Z'));
+            const streamedLog = await client.run('redirect-run-id').getStreamedLog({ fromStart: false });
+            jest.useRealTimers();
+
+            streamedLog.start();
+            // Wait some time to accumulate logs
+            await new Promise((resolve) => {
+                setTimeout(resolve, 1000);
+            });
+            await streamedLog.stop();
+
+            const loggerPrefix = c.cyan('redirect-actor-name runId:redirect-run-id -> ');
+            expect(logSpy.mock.calls).toEqual(
+                MOCKED_ACTOR_LOGS_PROCESSED.slice(1).map((item) => [loggerPrefix + item]),
+            );
+            logSpy.mockRestore();
+        });
```

Comment:
With `jest`, you can do a parametrized test like:

```typescript
const testCases = [
    { fromStart: true, expected: MOCKED_ACTOR_LOGS_PROCESSED.map...  },
    { fromStart: false, expected: MOCKED_ACTOR_LOGS_PROCESSED.slice(1).map... },
  ];

  test.each(testCases)('getStreamedLog $fromStart', ({ fromStart, expected }) => {
    expect(getStreamedLog(fromStart)).toBe(expected);
  });
```

Up to you if you want to use it, but might help with code duplication in those tests a little.